internal/test/wsu/wsu_test.go

```diff
@@ -77,14 +77,26 @@ ansible_winrm_server_cert_validation=ignore
 func TestWSU(t *testing.T) {
 	require.NotEmptyf(t, playbookPath, "WSU_PATH environment variable not set")
 
-	t.Run("VM specific tests", func(t *testing.T) {
-		testAllVMs(t)
-	})
+	// Run the WSU before applying hybrid overlay patch, expecting it to fail with a verbose error message
+	t.Run("Expect failure when hybrid overlay is not enabled", testWithoutHybridOverlay)
+
+	// Apply the hybrid overlay patch, and run the full VM test suite for each VM
+	err := framework.ApplyHybridOverlayPatch()
+	require.NoError(t, err, "Could not apply hybrid overlay patch")
```

Comment:
Do we need a delay after this given that the patch takes a while to take effect? Ideally we wait until the OVN pods are ready again or the hybrid annotations are present on all master nodes.

Reply:
Yes, we should probably be more thorough about waiting for things to finish setting up. I will come up with something

Reply:
I agree instead of introducing delay, it'd be better to go with the approach that @aravindhp suggested.